NuRadioReco/modules/channelGalacticNoiseAdder.py

```diff
@@ -91,132 +182,135 @@ def run(
+            S_per_bin = S * d_f
+
+            # calculate electric field per energy bin from the radiance per bin
+            E = np.sqrt(S_per_bin / (scipy.constants.c * units.m / units.s * scipy.constants.epsilon_0 * (
+                        units.coulomb / units.V / units.m))) / d_f
+
+            # assign random phases to electric field
+            noise_spectrum = np.zeros((3, freqs.shape[0]), dtype=np.complex128)
+            phases = np.random.uniform(0, 2. * np.pi, len(S))
+
+            noise_spectrum[1][passband_filter] = np.exp(1j * phases) * E
+            noise_spectrum[2][passband_filter] = np.exp(1j * phases) * E
+
+            channel_noise_spec = np.zeros_like(noise_spectrum)
+
+            for channel in station.iter_channels():
                 if detector.get_relative_position(station.get_id(), channel.get_id())[2] < 0:
-                    t_theta = geometryUtilities.get_fresnel_t_p(zenith, n_ice, 1)
-                    t_phi = geometryUtilities.get_fresnel_t_s(zenith, n_ice, 1)
-                    fresnel_zenith = geometryUtilities.get_fresnel_angle(zenith, n_ice, 1.)
-                    if fresnel_zenith is None:
-                        continue
+                    curr_t_theta = t_theta
+                    curr_t_phi = t_phi
+                    curr_fresnel_zenith = fresnel_zenith
                 else:
-                    t_theta = 1
-                    t_phi = 1
-                    fresnel_zenith = zenith
+                    curr_t_theta = 1
+                    curr_t_phi = 1
+                    curr_fresnel_zenith = zenith
+
+                antenna_pattern = self.__antenna_pattern_provider.load_antenna_pattern(
+                    detector.get_antenna_model(station.get_id(), channel.get_id()))
+                antenna_orientation = detector.get_antenna_orientation(station.get_id(), channel.get_id())
+
+                # calculate the phase offset in comparison to station center
+                # consider additional distance in air & ice
+                # assume for air & ice constant index of refraction
+                channel_pos_x, channel_pos_y = detector.get_relative_position(station.get_id(), channel.get_id())[
+                    [0, 1]]
+                channel_depth = abs(min(detector.get_relative_position(station.get_id(), channel.get_id())[2], 0))
+                sin_zenith = np.sin(zenith)
+                delta_phases = (
+                        2 * np.pi * freqs[passband_filter] / (scipy.constants.c * units.m / units.s) * n_air *
+                        (
+                                sin_zenith *
+                                (np.cos(azimuth) * channel_pos_x + channel_pos_y * np.sin(azimuth)) +
+                                channel_depth *
+                                ((n_ice / n_air) ** 2 + sin_zenith ** 2) /
+                                np.sqrt((n_ice / n_air) ** 2 - sin_zenith ** 2)
+                        )
+                )
```

Comment:
I think this is Moritz' change to account for the same signal in different antennas. This is definitely an upgrade we want to have, but as mentioned in #591 I don't think this is correct for deep in-ice antennas (I am okay with not fixing that for now, although maybe issuing a warning for channel depths over 10 m would be good?), and it also doesn't look right to me for above-surface antennas right now (that probably should be fixed and should be relatively straightforward).

Reply:
+ maybe a global sign flip (i.e., `-2 pi ...`)

Reply:
For the additional in-ice part I am getting `z * cos(theta) (n_ice/ n_air - sin^2(theta))`. But this could very well be wrong. Well it is wrong because it does not take into account refraction at the air-ice boundary but even if we ignore that it is probably wrong.

Reply:
I think you've got your signs wrong, `x cos \phi` should have the same sign as `y sin \phi`. And a positive phase implies a negative time delay in the NuRadio convention, so that is currently accurate as well. I get a result for the depth term that's different from both what's in there and your suggestion - maybe we should compare notes at some point!

Reply:
I get the sign from the rotation matrix around the z axis 
![image](https://github.com/user-attachments/assets/479952a4-0c6e-4aa3-8387-27cad04dda6c)

Reply:
I think (hope) I got my maths right ;) but I've resorted to just using `geometryUtilities.get_time_delay_for_direction` instead anyway. If you want to double check your maths I think the easiest is to just take the dot product of arrival direction and antenna position. And positive x at phi=0 should definitely give the same sign as positive y and phi=90 deg ;)

Reply:
You are right, I am just wondering where I am doing something wrong in my calculation? Do I have to rotated by `theta = - azimuth` because then since `cos(x) = cos(-x)` and `sin(x) = -sin(-x)` I would get a `+` in between the x and y term

Reply:
But I agree your solution with the utility is better (did you check for efficiency?).

Reply:
probably this is irrelevant

Reply:
> probably this is irrelevant

Like I mentioned somewhere below, everything is `get_antenna_response_vectorized` right now - if we can think of a way to speed this up by an order of magnitude or more we can look at other parts of this code for performance, but until then this is a waste of time, I think.